x/feegrant/keeper/keeper.go

```diff
@@ -49,6 +50,25 @@ func (k Keeper) GrantAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress,
 
 	store := ctx.KVStore(k.storeKey)
 	key := feegrant.FeeAllowanceKey(granter, grantee)
+
+	var exp *time.Time
+	existingGrant, err := k.getGrant(ctx, grantee, granter)
+	if err != nil && existingGrant.GetAllowance() != nil {
```

Comment:
this looks not correct. If `err != nil` the `existingGrant` can be nil and `existingGrant.GetAllowance` fails.